File: 2024/day1.rb

require "pry"                      
list1, list2 = [], []                                 
lines = File.open("day1.txt").read
.split("\n")   
.each do |line| 
  a,b = line.split(" ").map(&:strip).map(&:to_i)
  list1 << a
  list2 << b
end
#part1
difs = []    
list1.sort.each_with_index do |a, index|
  b = list2.sort[index]
  difs << (b - a).abs
end
#part2
counts = []
list1.each do |a|
  counts << a*list2.count(a)
end
#binding.pry
result1 = difs.sum
result2 = counts.sum
puts "part1: #{result1}" if result1   
puts "part2: #{result2}" if result2 

File: 2024/day1.js
const fs = require('fs');
const path = require('path'); 
const lines = fs.readFileSync(path.resolve(__dirname, 'day1.txt'), 'utf-8')
.split('\n');
const splits = lines.map((line, i) => line.split(' ').filter(a => a!=''));
const list1 = splits.map((split) => Number(split[0]));
const list2 = splits.map((split) => Number(split[1]));

// Part 1
const results1 = list1.sort().map((num, i) => {
    return Math.abs(num - list2.sort()[i]);
})
console.log("part1", results1.reduce((acc, item) => acc + item, 0));

// Part 2
const results2 = list1.map( a => {
    return a * list2.filter( n => n == a).length;
})
console.log("part2", results2.reduce((acc, item) => acc + item, 0));

